www/js/intro.js

```diff
@@ -112,10 +112,18 @@ angular.module('emission.intro', ['emission.splash.startprefs',
   };
 
   $scope.fetchItinerumSurvey = function(url) {
-     var userProfile = CommHelper.getUser();
+    (async () => {
+     let uuid="";
+     var step1 = new Promise(function(resolve, reject) {
+       resolve(CommHelper.getUser());
+     });
+     var step2 = step1.then(function(value){
+       uuid=value.user_id['$uuid'];
+     });
+     await step2;
```

Comment:
this should work. you could have also moved the entire `data_dict` setting code into the `then`, but this keeps the rest of the code clean instead of leading to a bunch of nested ifs. So I like it!